exporters/etw/include/opentelemetry/exporters/etw/etw_data.h

```diff
@@ -0,0 +1,536 @@
+#include <cstdio>
+#include <cstdlib>
+
+#include <mutex>
+
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/nostd/unique_ptr.h"
+
+#include "opentelemetry/common/key_value_iterable_view.h"
+#include "opentelemetry/trace/span.h"
+#include "opentelemetry/trace/span_context_kv_iterable_view.h"
+#include "opentelemetry/trace/span_id.h"
+#include "opentelemetry/trace/trace_id.h"
+#include "opentelemetry/trace/tracer_provider.h"
+
+#include "opentelemetry/sdk/trace/exporter.h"
+#include "opentelemetry/sdk/trace/recordable.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <unordered_map>
+
+#include "opentelemetry/exporters/etw/etw_provider_exporter.h"
+
+#include "opentelemetry/exporters/etw/utils.h"
+
+namespace core  = opentelemetry::core;
+namespace trace = opentelemetry::trace;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+
+namespace ETW
+{
+class Span;
+
+/// <summary>
+/// stream::Tracer class that allows to send spans to ETW
+/// </summary>
+class Tracer : public trace::Tracer
+{
+  /// <summary>
+  /// Parent provider of this Tracer
+  /// </summary>
+  trace::TracerProvider &parent;
+
+  /// <summary>
+  /// Provider Id (Name or GUID)
+  /// </summary>
+  std::string provId;
+
+  /// <summary>
+  /// Provider Handle
+  /// </summary>
+  ETWProvider::Handle provHandle;
+
+  /// <summary>
+  /// Encoding (Manifest, MessagePack or XML)
+  /// </summary>
+  ETWProvider::EventFormat encoding;
+
+  /// <summary>
+  /// ETWProvider is a singleton that aggregates all ETW writes.
+  /// </summary>
+  /// <returns></returns>
+  static ETWProvider &etwProvider()
+  {
+    static ETWProvider instance;  // C++11 magic static
+    return instance;
+  };
+
+public:
+  /// <summary>
+  /// Tracer constructor
+  /// </summary>
+  /// <param name="parent">Parent TraceProvider</param>
+  /// <param name="providerId">providerId</param>
+  /// <returns>Tracer instance</returns>
+  Tracer(trace::TracerProvider &parent,
+         nostd::string_view providerId     = "",
+         ETWProvider::EventFormat encoding = ETWProvider::EventFormat::ETW_MANIFEST)
+      : trace::Tracer(),
+        parent(parent),
+        provId(providerId.data(), providerId.size()),
+        encoding(encoding)
+  {
+    provHandle = etwProvider().open(provId, encoding);
+  };
+
+  /// <summary>
+  /// Start Span
+  /// </summary>
+  /// <param name="name">Span name</param>
+  /// <param name="options">Span options</param>
+  /// <returns>Span</returns>
+  virtual nostd::shared_ptr<trace::Span> StartSpan(
+      nostd::string_view name,
+      const common::KeyValueIterable &attributes,
+      const trace::SpanContextKeyValueIterable &links,
+      const trace::StartSpanOptions &options = {}) noexcept override
+  {
+    // TODO: support attributes
+    UNREFERENCED_PARAMETER(attributes);
+    UNREFERENCED_PARAMETER(links);
```

Comment:
This will be a problem for doxygen: https://stackoverflow.com/questions/55421894/how-do-i-doxygen-document-unnamed-parameters-of-a-function ... Since this module is Windows-specific, I believe `UNREFERENCED_PARAMETER` would be appropriate, esp. if we want to run doxygen on that code. Support for both parameters (attributes, links) will be added in a separate PR.

Reply:
I get your point.

We follow the pattern with unnamed parameters in other places, because it's true to [ISO CPP standards](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-unused). I also think that in most cases implementations of virtual methods in derived classes don't necessarily need documentation.